internal/services/compute/shared_image_versions_data_source.go

```diff
@@ -45,6 +46,13 @@ func dataSourceSharedImageVersions() *pluginsdk.Resource {
 				Computed: true,
 				Elem: &pluginsdk.Resource{
 					Schema: map[string]*pluginsdk.Schema{
+						"resource_id": {
+							Type:         pluginsdk.TypeString,
+							Optional:     true,
+							ForceNew:     true,
+							ValidateFunc: azure.ValidateResourceID,
```

Comment:
this is a Data Source so needs to be:

```suggestion
							Computed:     true,
```